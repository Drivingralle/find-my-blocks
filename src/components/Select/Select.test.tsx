import React from "react";
import renderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { Select, Option } from "./";

afterEach(cleanup);

it("render correctly", () => {
  const tree = renderer.create(
    <Select label="Test Label">
      <Option>Test</Option>
      <Option selected>Selected</Option>
      <Option value="withVal">With Val</Option>
    </Select>
  ).toJSON;
  expect(tree).toMatchSnapshot();
});

it("does not fires an event for `onChange` if no onChange set", () => {
  const onChange = jest.fn();
  const { getByLabelText } = render(
    <Select label="test">
      <Option value="1">1</Option>
      <Option value="2">2</Option>
    </Select>
  );
  const select = getByLabelText("test");
  fireEvent.change(select, {
    target: { value: "2" },
  });

  expect(onChange).not.toHaveBeenCalled();
});

it("fires an event for `onChange`", () => {
  const onChange = jest.fn();
  let v = "1";
  const { getByLabelText } = render(
    <Select
      label="test"
      onChange={(val) => {
        onChange();
        v = val;
      }}
    >
      <Option value="1">1</Option>
      <Option value="2">2</Option>
    </Select>
  );
  const select = getByLabelText("test");
  fireEvent.change(select, {
    target: { value: "2" },
  });

  expect(onChange).toHaveBeenCalled();
  expect(v).toEqual("2");
});
