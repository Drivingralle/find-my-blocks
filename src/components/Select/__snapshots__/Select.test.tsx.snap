// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should render correctly 1`] = `[Function]`;
